src/main/java/org/tugraz/sysds/runtime/controlprogram/context/ExecutionContext.java

```diff
@@ -367,7 +373,11 @@ public MatrixObject allocateGPUMatrixObject(String varName, long numRows, long n
 			e = e1;
 		}
 		try {
-			dim2 = validateDimensions(mo.getNumColumns(), numCols);
+			// ToDo: fix this hack
+			if(instName.equals("gpu_ucumk+*"))
```

Comment:
As you said, please get rid of this hack - there should be no opcodes in the execution context.